src/qtt/algorithms/functions.py

```diff
@@ -340,24 +341,46 @@ def cost_function(params):
     return fitted_parameters
 
 
+def damped_sine_wave(x_data, amplitude, decay_rate, power, frequency, phase, offset):
+    """
+    A sine wave with exponential decay of a power of x.
+    Function:
+    $$ amplitude * exp(-(x_data*decay_rate)**power) * sin(2pi*frequency * x_data + phase) + offset $$
+    Function parameters:
+
+    Args:
+        x_data (np.ndarray): array of independent variable values where the function needs to be evaluated.
+        amplitude: amplitude of the sine at x=0.
+        decay_rate (Hz): value of 1/x where the damping factor is 1/e
+        power (-): power of x_data*decay_rate inside the exponent.
+        frequency (Hz)
+        phase (radians): phase of the sine at x=0.
+        offset: constant offset of the function.
+
+    Returns:
+        np.ndarray with values of function for all x_data.
+    """
+    x = x_data
+    return amplitude * np.exp(-(x * decay_rate) ** power) * np.sin(2 * np.pi * frequency * x + phase) + offset
+
+
 def gauss_ramsey(x_data, params):
     """ Model for the measurement result of a pulse Ramsey sequence while varying the free evolution time, the phase
     of the second pulse is made dependent on the free evolution time. This results in a gaussian decay multiplied
     by a sinus.
     Function as used by T.F. Watson et all., example in qtt/docs/notebooks/example_fit_ramsey.ipynb
 
-    $$ gauss_ramsey = A * exp(-(x_data/t2s)**2) * sin(2pi*ramseyfreq * x_data - angle) +B  $$
+    $$ gauss_ramsey = A * exp(-(x_data/t2s)**2) * sin(2pi*ramsey_freq * x_data - angle) +B  $$
 
     Args:
-        x_data (array): the data for the input variable
-        params (array): parameters of the gauss_ramsey function, [A,t2s,ramseyfreq,angle,B]
+        x_data (numpy array): the data for the input variable
```

Comment:
Why write `numpy array` here and `array` in the output? Be consistent if possible.

Reply:
for the PyCharm type checker, `array` is different from `numpy array` (or `np.ndarray`), so I tried to put the proper type. We can use `np.ndarray` everywhere instead of `numpy array` (indeed good to be consistent there)